packages/sources/finnhub/src/endpoint/quote.ts

```diff
@@ -79,39 +78,52 @@ export const buildSymbol = ({
   return base
 }
 
-const requestTransform = (req: AdapterRequest<typeof inputParameters.validated>) => {
-  req.requestContext.data.base = req.requestContext.data.base.toUpperCase()
-  req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
-  req.requestContext.data.exchange = req.requestContext.data.exchange?.toUpperCase()
+const requestTransform =
+  (defaultExchange: string) => (req: AdapterRequest<typeof inputParameters.validated>) => {
+    req.requestContext.data.base = req.requestContext.data.base.toUpperCase()
+    req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
```

Comment:
Is the following use case an issue?
- in a request where the quote and exchange aren't set and base is not in the correct format, we can pass through this function with the `req.requestContext.data.exchange` being undefined.
- in `validateExchange`, the if statement would resolve to false due to the missing exchange param, so we wouldn't throw a 400
- this could result in the request being sent with bad params

Reply:
You're right that `exchange` would be undefined, and we intentionally skip the `validateExchange` check in that case. 

This is intentional because we also support queries that only provide the base, e.g. equities (`{ "base": "AAPL" }`). In this case, we don't want to do any symbol conversion to FX pairs, and just use the raw base passed in to query Finnhub. 

For FX, we always expect either 
1. a `base` and `quote` to be passed in (e.g. `{ "base": "EUR", "quote": "USD" }`), or 
2. a complete symbol (e.g. `{ "base": "FHFX:EUR_USD" }`).